src/lib/nvme_pt_ioctl.c

```diff
@@ -36,13 +36,15 @@
 #define ARRAY_SIZE(x) (sizeof(x) / sizeof(x[0]))
 #define MIN(x,y)  (((x - y) >> 63) & 1) ? x : y
 
-#define IO_BUFFER_LEN 2048
+#define MIN_IO_BUFFER_LEN 2048
 
 struct opal_device {
 	uint16_t comid;
 
-	uint8_t req_buf[IO_BUFFER_LEN];
-	uint8_t resp_buf[IO_BUFFER_LEN];
+	uint8_t *req_buf;
+	uint8_t *resp_buf;
+	uint64_t req_buf_size;
```

Comment:
do you expect buffer size to be 64bit in length?

Reply:
The response we parse from the tper properties are of type uint64_t. Just to avoid data type mismatch, I'm using uint64_t type here for req(resp)_buf_size.

Reply:
I'm ok to leave it as is, because I don't think it does any harm. It just caught my eye seeing 64B for buffer size